include/aspect/material_model/interface.h

```diff
@@ -674,6 +674,41 @@ namespace aspect
 
 
     /**
+     * Additional output fields for reaction rates to be added to
+     * the MaterialModel::MaterialModelOutputs structure and filled in the
+     * MaterialModel::Interface::evaluate() function.
+     *
+     * These reaction rates are only used in the operator_splitting nonlinear
+     * solver scheme, which allows it to solve reactions between compositional
+     * fields decoupled from the advection, and using a different time step size.
+     * In this case, they are used in addition to (and independent from) any
+     * reaction_terms that a material model defines, which are assembled as usual.
+     * For any other solver scheme, these values are ignored.
+     *
+     * In contrast to the reaction_terms, which are actual changes in composition
+     * rather than reaction rates, and assume equilibrium between the compositional
+     * fields, the reacion_rates defined here allow for reaction processes that are
+     * on shorter time scales than the advection, and disequilibrium reactions.
+     */
+    template <int dim>
+    class ReactionRateOutputs : public NamedAdditionalMaterialOutputs<dim>
+    {
+      public:
+        ReactionRateOutputs (const unsigned int n_points,
+                             const unsigned int n_comp);
+
+        virtual std::vector<double> get_nth_output(const unsigned int idx) const;
+
+        /**
+         * Reaction rates at the evaluation points passed to
+         * the instance of MaterialModel::Interface::evaluate() that fills
+         * the current object.
+         */
+        std::vector<std::vector<double> > reaction_rates;
```

Comment:
I think it works well the way it is right now. And if we find we need additional temperature rates in the material model, we can always add them later, that wouldn't break anything.